### ISSUE TITLE
Add iOS docs page

### ISSUE THREAD

**Issue by tfcporciuncula** [evant/kotlin-inject#241]
There's a great [Android](https://github.com/evant/kotlin-inject/blob/main/docs/android.md) docs page which helps a lot when getting started with the lib on the Android side. Once I was done with it and wanted to start on the iOS side, I was a bit lost and wished there was some iOS specific documentation as well.

Would it make sense to consider adding something like that? Even if all it said right now was "on iOS you should create the components by directly instantiating their generated implementation" it would already be helpful, but of course it would be even nicer if we can demonstrate some patterns and examples.

**Comment by evant** [comment]
Yeah sounds like a good idea. I don't have much expertise on iOS but pr's are welcome.
